src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataMember.cs

```diff
@@ -160,10 +165,16 @@ private sealed class CriticalHelper
             private bool _hasConflictingNameAndType;
             private DataMember? _conflictingMember;
 
+            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+               Justification = "The only use of this field is on the following property getter which is already " +
+                "annotated as RequiresUnreferencedCode.")]
             internal CriticalHelper(MemberInfo memberInfo)
             {
                 _emitDefaultValue = Globals.DefaultEmitDefaultValue;
                 _memberInfo = memberInfo;
+#pragma warning disable IL2026 // Methods annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
```

Comment:
?? What is this for?

Reply:
This is for the linker roslyn analyzer, I used both the analyzer and the CodeFixer when working on these changes and the analyzer doesn't look into the UnconditionalSuppressMessage attribute, so  you have to disable the warning like this for now. Since I only added the analyzer during my development of this, I can go ahead and remove the pragma warning for now, although I imagine we will eventually enable the analyzer on the whole repo which will then require to either re-add this, or change the analyzer to look into the attribute.

Reply:
The analyzer should look for the attribute.

cc @agocke

Reply:
The version of Roslyn in runtime should support `UnconditionalSuppressMessage` as of a day or so ago. You should be able to add it and have things work as you expect.